File: src/main/java/cn/labzen/web/ap/processor/ReadSourceProcessor.java
package cn.labzen.web.ap.processor;

import cn.labzen.web.ap.internal.context.ControllerContext;
import cn.labzen.web.ap.internal.element.ElementClass;

import javax.lang.model.type.TypeMirror;

public final class ReadSourceProcessor implements InternalProcessor {

  @Override
  public void process(ControllerContext context) {
    String className = context.getSource().getSimpleName().toString() + context.getApc().config().classNameSuffix();
    String pkg = context.getApc().elements().getPackageOf(context.getSource()).getQualifiedName().toString();

    TypeMirror implementTypes = context.getSource().asType();

    context.setRoot(new ElementClass(className, pkg, implementTypes));
  }

  @Override
  public int priority() {
    return PRIORITY_READ_SOURCE;
  }
}

File: src/main/kotlin/cn/labzen/web/ap/processor/ReadSourceProcessor.kt
package cn.labzen.web.ap.processor

import cn.labzen.web.ap.internal.context.ControllerContext
import cn.labzen.web.ap.internal.element.ElementClass
import cn.labzen.web.ap.processor.InternalProcessor.Companion.PRIORITY_READ_SOURCE

class ReadSourceProcessor : InternalProcessor {

  override fun process(context: ControllerContext) {
    with(context) {
      val className = source.simpleName.toString() + apc.config.classNameSuffix()
      val pkg = apc.elementUtils.getPackageOf(source).qualifiedName.toString()

      val implements = source.asType()

      root = ElementClass(className, pkg, implements)
    }
  }

  override fun priority(): Int = PRIORITY_READ_SOURCE
}